app/services/employees/csv_parser.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Employees
+  class CsvParser < ApplicationService
+    attr_reader :attachment, :company, :approved_users
+
+    USER_PROPS = %i[email first_name last_name role phone].freeze
+
+    def initialize(attachment, company)
+      super()
+      @attachment = attachment
+      @company = company
+      @approved_users = []
+    end
+
+    def call
+      CSV.foreach(attachment, headers: true, header_converters: :symbol, col_sep: ';') do |row|
+        columns ||= row.headers.intersection(USER_PROPS)
```

Comment:
We need to validate csv file on correct headers, and say this to user, so I'd return status of processing CSV file

Reply:
Also think about delayed job, during creating/processing files, cause it could be pretty time consuming operation and you'll hang server/browser and it's bad idea :)